### PR TITLE
Fix goroutine leak in startVMM

### PR DESCRIPTION
**Pull request by bduffany**
*Issue #, if available:*

N/A

*Description of changes:*

The statement `<-ctx.Done()` without a `select {}` wrapping it will cause a goroutine leak if `ctx` is `context.Background()` or `context.TODO()` (or any other context that doesn't get cancelled, intentionally or not).

This change allows the goroutine to exit if the VM exits on its own, in addition to waiting for `ctx.Done()`, preventing a goroutine leak in case people do pass in a context that doesn't get canceled, but are otherwise managing the VM lifecycle correctly.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: machine.go [hunk 1]
```diff
@@ -613,10 +613,16 @@ func (m *Machine) startVMM(ctx context.Context) error {
...
 
-	// This goroutine is used to kill the process by context cancelletion,
+	// This goroutine is used to kill the process by context cancellation,
 	// but doesn't tell anyone about that.
 	go func() {
-		<-ctx.Done()
+		select {
+		case <-ctx.Done():
+			break
+		case <-m.cleanupCh:
```
Comment (Kern--) [inline machine.go:622]: nitpick: `m.cleanupCh` works, but to to make it clear that what we really care about is whether the VM exited, can you use `m.exitCh`?

**Comment by Kern--** [comment]
Thanks, just one minor comment :)

**Comment by Kern--** [comment]
Actually, one more, can you please add a signed off line to your commit message like:

```
Signed-off-by: Firstname Lastname <email@provider.com>
```

File: machine.go (hunk 1 above)
Comment (bduffany) [reply machine.go]: Good call; done

**Comment by bduffany** [comment]
> can you please add a signed off line to your commit message

Done

**Comment by bduffany** [comment]
I'm not sure how to interpret the failing CI tests; it looks like maybe TestLoadSnapshot timed out. Is that a known issue @Kern-- or is it likely this PR could've caused it?

**Review by Kern-- — APPROVED** [review]

**Review by austinvazquez — APPROVED** [review]

**Pull request merged**